### DISCUSSION TITLE
Table results disappear after I apply filters and refresh the page

### DISCUSSION THREAD

**Discussion by maxacarvalho** [filamentphp/filament#8112] [Help]
Hi,

Perhaps this is a long shot since I don't have an example app to share but I'll try asking, just in case someone already spotted it.

The thing is, I have list pages, some simple and some more complex. Those have table filters, pretty  basic.
Whenever I apply a filter and refresh the page the results are gone, I only see a "No records" message.
Then, if I click the "Reset filters" link the results reappear. But if I refresh the page, gone again.

Note: it only happens with table filters. With the "search" input field everything works as expected.

Did anyone saw that happening?

**Comment by leandrocfe** [comment]
share the code please

**Reply by maxacarvalho** [reply]
I did some deeper research, it has to do with the issue reported [here](https://github.com/filamentphp/filament/issues/7129).

As a temporary solution, add the following snippet to the list page.
But be aware that although it fixes the results disappearing, it messes with the filter indicator.

```php
public function boot(): void
{
    if ($this->tableFilters) {
        $this->replaceNullValues($this->tableFilters);
    }
}

protected function replaceNullValues(array &$data): void
{
    foreach ($data as &$value) {
        if (is_array($value)) {
            $this->replaceNullValues($value);
        } elseif ($value === "null") {
            $value = null;
        }
    }
}
```
